erpnext/www/book-appointment/index.js

```diff
@@ -0,0 +1,214 @@
+frappe.ready(async () => {
```

Comment:
We can work on a refactor of this part and write a class based approach.. Date sharing, hence state management will be easy and clean